### PR TITLE
[MRG+2] Mep12 ticks and spines

### PR DESCRIPTION
**Pull request by NelleV**
I've merged two examples and wrote documentation for a bunch of others.

refs #7206

### CODE REVIEW

File: examples/ticks_and_spines/spines_demo.py [hunk 1]
```diff
@@ -26,7 +30,18 @@
 ax1.yaxis.set_ticks_position('left')
 ax1.xaxis.set_ticks_position('bottom')
 
+
+ax2.plot(x, y)
+
+# Only draw spine between the y-ticks
+ax2.spines['left'].set_bounds(-1, 1)
```
Comment (tacaswell) [inline examples/ticks_and_spines/spines_demo.py]: Well, I learned something...

**Comment by tacaswell** [comment]
Please backport to 2.0.0-doc

File: examples/ticks_and_spines/spines_demo.py [hunk 2]
```diff
@@ -26,7 +30,18 @@
 ax1.yaxis.set_ticks_position('left')
 ax1.xaxis.set_ticks_position('bottom')
 
+
```
Comment (anntzer) [inline examples/ticks_and_spines/spines_demo.py]: no need for two empty lines?  (there's only one between ax0 and ax1)

File: examples/ticks_and_spines/tick-locators.py [hunk 3]
```diff
@@ -1,4 +1,8 @@
 """
+=============
+Tick locators
+=============
+
 Show the different tick locators
```
Comment (anntzer) [inline examples/ticks_and_spines/tick-locators.py]: Add a dot.

File: examples/ticks_and_spines/spines_demo.py [hunk 4]
```diff
@@ -1,8 +1,12 @@
...
-Basic demo of axis spines.
-
-This demo compares a normal axes, with spines on all four sides, and an axes
-with spines only on the left and bottom.
+======
+Spines
+======
+
+This demo compares :
+    - normal axes, with spines on all four sides
```
Comment (anntzer) [inline examples/ticks_and_spines/spines_demo.py]: Add dots (or commas) at the end of each item.

File: examples/ticks_and_spines/spines_demo.py [hunk 5]
```diff
@@ -1,8 +1,12 @@
 """
-Basic demo of axis spines.
-
-This demo compares a normal axes, with spines on all four sides, and an axes
-with spines only on the left and bottom.
+======
+Spines
+======
+
+This demo compares :
```
Comment (anntzer) [inline examples/ticks_and_spines/spines_demo.py]: no space before the colon (English typography :-))

**Comment by NelleV** [comment]
I fixed the blank spaces/lines issues.

File: examples/ticks_and_spines/spines_demo.py [hunk 6]
```diff
@@ -3,10 +3,10 @@
...
 ======
 
-This demo compares :
-    - normal axes, with spines on all four sides
-    - an axes with spines only on the left and bottom.
-    - an axes using custom bounds to limit the extent of the spine
+This demo compares:
+    - normal axes, with spines on all four sides;
+    - an axes with spines only on the left and bottom;
+    - an axes using custom bounds to limit the extent of the spine;
```
Comment (anntzer) [inline examples/ticks_and_spines/spines_demo.py]: not at the end of the last item...

Comment (NelleV) [inline examples/ticks_and_spines/spines_demo.py]: As a side note, when it starts being that nitpicky, it is much faster to fix it yourself before merging.

Comment (NelleV) [inline examples/ticks_and_spines/spines_demo.py]: FYI, I didn't even bother doing it on my computer: I just did it through the github API.

Comment (anntzer) [inline examples/ticks_and_spines/spines_demo.py]: Ah, I didn't know I could do it directly there... good point.

**Comment by anntzer** [comment]
Thanks.

**Comment by dstansby** [comment]
Backported to `2.0.0-doc` via. https://github.com/matplotlib/matplotlib/commit/6750f68807f96a43ec143c357ed79eb7294b7ed5

**Pull request merged**